File: ListsService/src/main/java/com/filmmaster/listsservice/model/dto/UserInfo.java
package com.filmmaster.listsservice.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserInfo {

    String email;
    String name;
    String password;

    public UserInfo encryptPassword() {
        String password = new BCryptPasswordEncoder().encode(this.password);
        return new UserInfo(email, name, password);
    }

}

File: streamster_app/lib/common/service/dto/user_info.dart
class UserInfo {
  String firstName;
  String lastName;
  String email;
  String password;
  String avatar;

  UserInfo(this.firstName, this.lastName, this.email, this.password, this.avatar);

  Map toJson() =>
      {
        'firstName': firstName,
        'lastName': lastName,
        'email': email,
        'password': password,
        'avatar': avatar
      };
}
